lib/src/data/repository/movie_remote_repository.dart

```diff
@@ -0,0 +1,108 @@
+import 'dart:convert';
+
+import 'package:either_dart/either.dart';
+
+import '../../core/util/constants.dart';
+import '../../core/util/either_string_t.dart';
+import '../../domain/entity/movie_entity.dart';
+import '../../domain/repository/movie_repository_i.dart';
+import '../datasource/mappers/movie_mapper.dart';
+import '../datasource/remote/api_service.dart';
+import '../model/movie_data.dart';
+
+class MovieRemoteRepository implements IMovieRepository {
```

Comment:
If you can see here, you are replicating the same code 4 times, the only thing that is change in is the endpoint that you use, so instead to made for similar methods, you can use only one and as a parameter you can select which endpoint use

Reply:
fixed